ow_regolith/include/ServiceClientFacade.h

```diff
@@ -0,0 +1,117 @@
+// The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+// Research and Simulation can be found in README.md in the root directory of
+// this repository.
+
+// ServiceClientFacade wraps ros::ServiceClient and implements automatic
+// reconnection for when persistent clients drop their connection.
+
+#ifndef SERVICE_CLIENT_FACADE_H
+#define SERVICE_CLIENT_FACADE_H
+
+#include <memory>
+
+#include <ros/ros.h>
+
+namespace ow_regolith {
+
+class ServiceClientFacade
+{
+public:
+  ServiceClientFacade() : m_node_handle(nullptr), m_persistent(false)
+  {
+    // do nothing
+  }
+  ServiceClientFacade(const ServiceClientFacade&) = delete;
```

Comment:
Please comment on why the copy constructor and assignment operator are explicitly prohibited, rather than just not implementing them.

Reply:
This was in response to my comment when these weren't there.  I believe in automatically generated methods being implemented, defaulted, or deleted, i.e. being explicit about the semantics here.

The pre-C++-11 way to disallow copy, etc was to declare these private and then not implement them.

In this case I'm not sure if copying needs to be prohibited, but it doesn't matter until a use case arises.

Reply:
The underline reason to delete both copy and assignment is that they both require copying an instance, and supporting that safely would require testing of the ROS service API. Because I cannot think of a use case for this anyways, it will go untested for now. We can always add this feature down the line if it's needed.